vimiv/qt/__init__.py

```diff
@@ -0,0 +1,76 @@
+# vim: ft=python fileencoding=utf-8 sw=4 et sts=4
+
+# This file is part of vimiv.
+# Copyright 2017-2023 Christian Karl (karlch) <karlch at protonmail dot com>
+# License: GNU GPL v3, see the "LICENSE" and "AUTHORS" files for details.
+
+"""Wrapper module around qt to support various python wrapper versions."""
+
+import importlib
+import enum
+import os
+
+
+class WRAPPERS(enum.Enum):
+    """List of supported python Qt wrappers.
+
+    Order matters as this is the order in which we check for them being installed if the
+    wrapper is not defined by the user explicitly.
+    """
+
+    PyQt5 = "PyQt5"
+    PyQt6 = "PyQt6"
+    PySide6 = "PySide6"
+
+
+_WRAPPERS_PYQT = [WRAPPERS.PyQt5, WRAPPERS.PyQt6]
+_WRAPPER_NAMES = [wrapper.value for wrapper in WRAPPERS]
+_WRAPPER_NAMES_STR = ", ".join(_WRAPPER_NAMES)
+
+WRAPPER = None
+
+
+def _select_wrapper():
+    """Select a python Qt wrapper version based on environment variable and fallback."""
+    env_var = "VIMIV_QT_WRAPPER"
```

Comment:
Why do you introduce a custom env variable instead of relying on the *standardized* `QT_SELECT=5/6`?

Reply:
How would we differentiate between `PyQt6` and `PySide6`?

Reply:
Just take whichever is available. I doubt that users have both installed. And if so, why should they care which one is used?

Reply:
I care for testing both without changing the source code all the time :smile: I do think it is nice to expose this option in any case, but we could split it into `QT_SELECT` and `USE_PYSIDE`, if you think this has some benefit? Personally, I find one easier than two though.

Reply:
True, for development/testing/debugging this would certainly make sense to explicitly choose between QT and Side. I just think that if there is a standard, it would make sense to follow it. What about having only `QT_SELECT` but allow not only the values 5/6 but also the explicit options PyQt6/PySide6?

Reply:
I like this compromise :+1: So our `QT_SELECT` would then have:
* 5 -> PyQt5
* 6 -> PyQt6 or PySide6 with a preference that we set
* all three explicit options as currently done by `VIMIV_QT_WRAPPER`